pkg/secrets-store/provider_client_test.go

```diff
@@ -511,3 +511,38 @@ func TestPluginClientBuilder_HealthCheck(t *testing.T) {
 
 	wg.Wait()
 }
+
+func TestIsMaxRecvMsgSizeError(t *testing.T) {
+	cases := []struct {
+		name string
+		// inputs
+		err error
+		// expectations
+		expectedBool bool
+	}{
+		{
+			name:         "not resource exhausted error",
+			err:          errors.New("failed to mount"),
+			expectedBool: false,
+		},
+		{
+			name:         "resource exhausted error because of quota propagation",
+			err:          status.Errorf(codes.ResourceExhausted, "grpc: received message larger than max length allowed on current machine"),
+			expectedBool: false,
+		},
+		{
+			name:         "resource exhausted error because message larger than max length",
+			err:          status.Errorf(codes.ResourceExhausted, "grpc: received message larger than max"),
+			expectedBool: true,
+		},
```

Comment:
Also needs a test case where it is an `codes.ResourceExhausted` error but from something other than the internal grpc system (example, GCP APIs return codes.ResourceExhausted for user-quota errors):

```
		{
			name: "generic resource exhausted error",
			err:  status.Errorf(codes.ResourceExhausted, "user quota exceeded"),
			want: false,
		},
```